### DISCUSSION TITLE
TERM for ncurses

### DISCUSSION THREAD

**Discussion by fswarbrick** [zopencommunity/meta#814] [Porting]
So in the zopen/usr/local/zopen/ncurses/ncurses directory there is an .appenv file that has the following:
```
if [ "${_BPX_TERMPATH-x}" = "OMVS" ]; then
  export TERM=dumb
else
  export TERM=xterm
fi
```
The problem is that this essentially "turns off color support" for any session in which this is invoked.  On my system I've changed this to the following:
```
if [ "${_BPX_TERMPATH-x}" = "OMVS" ]; then
  export TERM=dumb
else
  if [[ $TERM != *xterm* && $TERM != *tmux* ]]; then
    export TERM=xterm
  fi
fi
```
This likely doesn't handle everything, but it does handle a regular terminal (SSH) session, as well as sessions created by the tmux and screen applications.  Should something like this be added to the zopen tool?

**Comment by IgorTodorovskiIBM** [comment]
> So in the zopen/usr/local/zopen/ncurses/ncurses directory there is an .appenv file that has the following:
> 
> ```
> if [ "${_BPX_TERMPATH-x}" = "OMVS" ]; then
>   export TERM=dumb
> else
>   export TERM=xterm
> fi
> ```
> 
> The problem is that this essentially "turns off color support" for any session in which this is invoked. On my system I've changed this to the following:
> 
> ```
> if [ "${_BPX_TERMPATH-x}" = "OMVS" ]; then
>   export TERM=dumb
> else
>   if [[ $TERM != *xterm* && $TERM != *tmux* ]]; then
>     export TERM=xterm
>   fi
> fi
> ```
> 
> This likely doesn't handle everything, but it does handle a regular terminal (SSH) session, as well as sessions created by the tmux and screen applications. Should something like this be added to the zopen tool?

Good suggestion, I actually prefer using xterm-256color and that setting would override it. Would you like to contribute a PR?

**Comment by fswarbrick** [comment]
To which repo would I do a PR for?

**Reply by IgorTodorovskiIBM** [reply]
It would be under ncursesport. This file: https://github.com/ZOSOpenTools/ncursesport/blob/main/buildenv#L41

**Reply by fswarbrick** [reply]
Ah.  I didn't see .appenv there, so figured that wasn't the right one.  I will do the PR.
